src/ToData.purs

```diff
@@ -49,6 +67,8 @@ instance (ToData k, ToData v) => ToData (Map k v) where
     where
     entries = Map.toUnfoldable mp :: Array (k /\ v)
 
+-- Note that nothing prevents the denominator from being zero, we could provide
```

Comment:
It looks like the property of not having zero denominator is not an invariant of `Ratio`, so we shouldn't do that. (Why though? I don't know)

I think we can remove the comment.

Reply:
So it does seem like `Ratio` doesn't care about this property, and it's down to the user to do whatever they wish, in which case a `Newtype` as you previously suggested could be best. I think the main thing is to replicate whatever Plutus does

Reply:
Happy to leave this for a separate PR btw. Just to get this merged in asap.